### PR TITLE
filter posts

### PR DESCRIPTION
**Pull request by candelarialopezz**

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -20,8 +20,7 @@ def show; end
 
   def index
     posts = current_user.followed_posts
-    sort_by = params[:sort_by]
-    @posts = define_posts(sort_by, posts)
+    @posts = Post.define_posts(posts, params)
```
Comment (sebastian-reolon) [inline app/controllers/posts_controller.rb:23]: Let's avoid creating a method inside the Post model. Move the whole define_posts logic inside the controller.

File: app/models/post.rb [hunk 2]
```diff
@@ -25,6 +25,65 @@ class Post < ApplicationRecord
...
+    posts = filter_posts(params, posts)
+    sort_posts(params[:sort_by], posts)
+  end
+
+  def self.filter_posts(params, posts)
+    posts = filter_author(posts, params[:author_filter])
+    posts = filter_title(posts, params[:title_filter])
+    posts = filter_body(posts, params[:body_filter])
+    filter_date(posts, params[:date_filter])
+  end
```
Comment (sebastian-reolon) [inline app/models/post.rb:38]: These methods would be present on the controller

File: app/models/post.rb [hunk 3]
```diff
@@ -25,6 +25,65 @@ class Post < ApplicationRecord
...
+    case date_filter
+    when 'Last day'
+      posts = posts.where('published_at >= ? ', now - 1.day)
+    when 'Last week'
+      posts = posts.where('published_at >= ?', now.beginning_of_week)
+    when 'Last month'
+      posts = posts.where('published_at >= ?', now.beginning_of_month)
+    end
+    posts
+  end
```
Comment (sebastian-reolon) [inline app/models/post.rb:73]: And this would stay in the Post model, but defined as scopes

File: app/javascript/dropdown.js [hunk 4]
```diff
@@ -3,21 +3,42 @@ $(document).ready(function() {
...
+
+    if ($(this).hasClass('dropdown-item')) {
+      var sortBy = $(this).data('sort-by');
+    } else if ($(this).hasClass('btn-apply')) {
+      var sortBy = $('.btn-dropdown').text();
+    } 
+
+    const authorFilter =$('.author-filter').val();
+    const titleFilter =$('.title-filter').val();
+    const bodyFilter =$('.body-filter').val();
```
Comment (daizuoheng) [inline app/javascript/dropdown.js:17]:
```suggestion
    const authorFilter = $('.author-filter').val();
    const titleFilter = $('.title-filter').val();
    const bodyFilter = $('.body-filter').val();
```

File: app/models/post.rb [hunk 5]
```diff
@@ -25,6 +25,38 @@ class Post < ApplicationRecord
...
+  def self.sort_posts(sort_by, posts)
+    case sort_by
+    when 'Trending'
+      posts.by_trending
+    when 'Number of likes'
+      posts.by_likes
+    else
+      posts.by_publishing_date
+    end
+  end
```
Comment (sebastian-reolon) [inline app/models/post.rb:58]: You can move this to a scope as well

**Review by sebastian-reolon — APPROVED** [review]

**Review by daizuoheng — APPROVED** [review]

**Pull request merged**